tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MacroSuite.scala

```diff
@@ -0,0 +1,169 @@
+package scala.meta.tests.parsers.dotty
+
+import scala.meta.tests.parsers._
+import scala.meta._
+
+class MacroSuite extends BaseDottySuite {
+
+  implicit val parseBlock: String => Stat = code => blockStat(code)(dialects.Dotty)
+
+  /** All examples based on dotty documentation:
+   *  https://dotty.epfl.ch/docs/reference/metaprogramming/toc.html
+   *  https://dotty.epfl.ch/docs/reference/metaprogramming/macros.html
+   *  https://dotty.epfl.ch/docs/reference/metaprogramming/tasty-reflect.html
+   *
+   *  '{ ... } OR 'ident - QuotedMacroExpr
+   *  '[ ... ] - QuotedMacroType
+   *  ${ ... } OR $ident - SplicedMacroExpr
+   *  symbols 'ident are not supported in dotty
+   */
+  test("parse-single-quote-character") {
+    runTestAssert[Stat]("val a = 'c'")(
+      Defn.Val(Nil, List(Pat.Var(tname("a"))), None, Lit.Char('c'))
+    )
+  }
+
+  test("macro-quote-expr") {
+    runTestAssert[Stat]("'{ Int.MinValue }")(
+      Term.QuotedMacroExpr(List(Term.Select(tname("Int"), tname("MinValue"))))
+    )
+    runTestAssert[Stat]("'{ (x: T) => 'x }")(
+      Term.QuotedMacroExpr(List(Term.Function(List(tparam("x", "T")), Lit.Symbol('x))))
+    )
+    runTestAssert[Stat]("f('{ 2 })")(
+      Term.Apply(tname("f"), List(Term.QuotedMacroExpr(List(Lit.Int(2)))))
+    )
+    runTestAssert[Stat]("if (b) '{ true } else '{ false }")(
+      Term.If(
+        tname("b"),
+        Term.QuotedMacroExpr(List(Lit.Boolean(true))),
+        Term.QuotedMacroExpr(List(Lit.Boolean(false)))
+      )
+    )
+    runTestAssert[Stat]("'{ val y = a; ${ env } }", assertLayout = false)(
+      Term.QuotedMacroExpr(
+        List(
+          Defn.Val(Nil, List(Pat.Var(tname("y"))), None, tname("a")),
+          Term.SplicedMacroExpr(List(tname("env")))
+        )
+      )
+    )
+    runTestAssert[Stat]("x match { case '{ a } => 1 }", assertLayout = false)(
+      Term.Match(
+        tname("x"),
+        List(Case(Pat.Macro(Term.QuotedMacroExpr(List(tname("a")))), None, Lit.Int(1)))
+      )
+    )
+  }
+
+  test("macro-quote-type") {
+    runTestAssert[Stat]("'[ Map[Int, String] ]")(
+      Term.QuotedMacroType(Type.Apply(pname("Map"), List(pname("Int"), pname("String"))))
+    )
+    runTestAssert[Stat]("'[ List[${ summon[Type[T]] }] ]")(
+      Term.QuotedMacroType(
+        Type.Apply(
+          pname("List"),
+          List(
+            Type.Macro(
+              Term.SplicedMacroExpr(
+                List(
+                  Term.ApplyType(tname("summon"), List(Type.Apply(pname("Type"), List(pname("T")))))
+                )
+              )
+            )
+          )
+        )
+      )
+    )
+    runTestAssert[Stat]("'[ Show[$tp] ]")(
+      Term.QuotedMacroType(Type.Apply(pname("Show"), List(pname("$tp"))))
```

Comment:
Should in not be `SplicedMacroExpr` ?

Reply:
Yes it should be `SplicedMacroExpr`, good catch :+1:

Reply:
fixed